source/node_test.go

```diff
@@ -383,3 +420,98 @@ func testNodeSourceEndpoints(t *testing.T) {
 		})
 	}
 }
+
+func testNodeEndpointsWithIPv6(t *testing.T) {
+	for _, tc := range []struct {
+		title              string
+		annotationFilter   string
+		labelSelector      string
+		fqdnTemplate       string
+		nodeName           string
+		nodeAddresses      []v1.NodeAddress
+		labels             map[string]string
+		annotations        map[string]string
+		exposeInternalIPv6 bool // default to true for this version. Change later when the next minor version is released.
```

Comment:
In this array there are no test cases covering `exposeInternalIPv6: true`. Is this intentional?

Reply:
All of the tests above are having exposeInternalIPv6 as true and since that is the default behavior, I omitted the test but if the above comment that you posted holds true i.e. we want internal IPs to be exposed if there are only internal IPs then would need to add a test to check that case.

Reply:
@hjoshi123 Yes, that's right. A test for this case is needed.

Reply:
Added the test @mloiseleur